### PR TITLE
LogIn

### PR DESCRIPTION
**Pull request by lamont1008**

### CODE REVIEW

File: app/http/controllers/AuthController.ts [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
+   * Handles the login request
+   *
+   * @param req {Request} Express request object
+   * @param res {Response} Express response object
+   * @param next {NextFunction} Express NextFunction (used for middleware)
+   */
+
+  async Login(req: Request, res: Response, next: NextFunction) {
+    try {
+      const tok = await AuthService.ldapJs(
```
Comment (JungAn2) [inline app/http/controllers/AuthController.ts:15]: instead of tok use the whole word token

File: app/services/AuthService.ts [hunk 2]
```diff
@@ -67,4 +72,99 @@ export const AuthService = {
 
     return payload;
   },
+
+  /**
+   *
```
Comment (JungAn2) [inline app/services/AuthService.ts:77]: Add a description of what the function does

File: app/services/AuthService.ts [hunk 3]
```diff
@@ -67,4 +72,99 @@ export const AuthService = {
...
+   *
+   * @param {String} token Authorization token attached to the HTTP header
```
Comment (JungAn2) [inline app/services/AuthService.ts:79]: param changes from token to uid

File: app/services/AuthService.ts [hunk 4]
```diff
@@ -67,4 +72,99 @@ export const AuthService = {
...
+      config.app.secret
+    );
+
+    const found = await prisma.users.findFirst({
+      where: {
+        uid: uid,
+      },
+    });
+    if (found === null) {
+      // will create JWT token if user id is not found
```
Comment (JungAn2) [inline app/services/AuthService.ts:101]: The comment here is not correct. It doesn't create jwt token but creates a row inside the database

File: app/services/AuthService.ts [hunk 5]
```diff
@@ -67,4 +72,99 @@ export const AuthService = {
...
+
+    client.on("error", (err) => {
+      //console.log("test")
+      // handle connection error
+    });
+
+    var user = "";
+
+    await client.bind(
+      "uid=" + uid + ",ou=users,dc=hslinux,dc=Salisbury,dc=edu",
```
Comment (JungAn2) [inline app/services/AuthService.ts:138]: Not sure about this one, but maybe putting the dn to the .env might be better way

File: app/services/AuthService.ts [hunk 6]
```diff
@@ -67,4 +72,99 @@ export const AuthService = {
...
+
+        client.unbind();
+
+        console.log(user);
+        //Will sign the JSON webtoken if no error
+        if (err == null) {
+          console.log("pass");
+          user = await this.jwtSign(uid);
+
+          // return user
```
Comment (JungAn2) [inline app/services/AuthService.ts:151]: Not returning the user here so delete this comment

**Review by JungAn2 — CHANGES_REQUESTED** [review]
The code looks good but the comments and documentation does need to change.

File: app/http/controllers/AuthController.ts [hunk 7]
```diff
@@ -1,21 +1,24 @@
...
+   * @param next {NextFunction} Express NextFunction (used for middleware)
+   */
+
+  async Login(req: Request, res: Response, next: NextFunction) {
+    try {
+      const token = await AuthService.ldapJs(
+        req.body.username,
+        req.body.password
+      );
+      res.send(token);
```
Comment (wrathofrathma) [inline app/http/controllers/AuthController.ts:19]: So I think this is sending to the client, just the plain text token.

Do we want to send the token in a JSON object? Actually asking, up to you guys

**Pull request merged**